File: codeforces/1800~1899/1875/d.cc
#include <bits/stdc++.h>

using namespace std;

using i64 = long long;

void solve() {
  int n = 0;
  cin >> n;
  vector<int> a(n);
  for (int i = 0; i < n; ++i) cin >> a[i];

  map<int, int> freq;
  for (int i = 0; i < n; ++i) freq[a[i]]++;

  int mex = 0;
  while (freq.count(mex)) {
    mex++;
  }

  constexpr int INF = numeric_limits<int>::max();
  vector<i64> dp(mex+1, INF);
  dp.back() = 0;
  for (int i = mex; i >= 0; --i) {
    for (int j = 0; j < i; ++j) {
      dp[j] = min(dp[j], dp[i] + i * (freq[j] - 1) + j);
    }
  }
  // for (int i = 0; i <= mex; ++i) {
  //   cout << dp[i] << " ";
  // }
  // cout << endl;

  cout << dp[0] << endl;
}

int main(void) {
  ios::sync_with_stdio(false);
  cin.tie(0);

  int T = 0;
  cin >> T;
  for (int t = 0; t < T; ++t) {
    solve();
  }
  return 0;
}

File: codeforces/1800~1899/1875/d.rs
use std::io::{stdin, stdout, BufWriter, Write};
use std::str::FromStr;
use std::fmt::Debug;
use std::cmp::min;
use std::collections::BTreeMap;

fn read_int<T: FromStr>() -> T where <T as FromStr>::Err: Debug {
    let mut s = String::new();
    stdin().read_line(&mut s).unwrap();
    s.trim().parse::<T>().expect("invalid input")
}

fn read_vec<T: FromStr>() -> Vec<T> where <T as FromStr>::Err: Debug {
    let mut s = String::new();
    stdin().read_line(&mut s).unwrap();
    s.split(" ").map(|x| x.trim().parse().expect("invalid input")).collect()
}

fn solve() {
    let stdout = stdout();
    let mut writer = BufWriter::new(stdout.lock());

    let n = read_int::<usize>();
    let a = read_vec::<i64>();
    let mut freq = BTreeMap::new();

    for i in 0..n {
        freq.entry(&a[i]).and_modify(|c| *c += 1).or_insert(1);
    }

    let mut mex = 0;
    while freq.contains_key(&mex) {
        mex += 1;
    }

    let inf = i32::MAX;
    let mut dp = vec![inf as i64; (mex+1) as usize];
    dp[mex as usize] = 0;
    for i in (0..=mex).rev() {
        for j in 0..i {
            let j_count = *freq.get(&j).unwrap();
            dp[j as usize] = min(dp[j as usize], dp[i as usize] + i * (j_count - 1) + j);
        }
    }

    writeln!(writer, "{}", dp[0]).ok();
}

fn main() {
    let n = read_int::<i32>();
    for _t in 0..n {
        solve();
    }
}
